### ISSUE TITLE
OAuth2Options reuse the same JwtOptions instance

### ISSUE THREAD

**Issue by bitkid** [eclipse-vertx/vertx-auth#676]
version 4.5.7

this test fails

```
public void test() {
    OAuth2Options options1 = new OAuth2Options();
    options1.getJWTOptions().setIssuer("one");

    OAuth2Options options2 = new OAuth2Options();
    options2.getJWTOptions().setIssuer("two");

    assertEquals(options1.getJWTOptions().getIssuer(), "one");
}
```

**Comment by bitkid** [comment]
JWTAuthOptions may have the same problem

**Comment by tsegismont** [comment]
Thanks for reporting this, we'll look into it.

**Comment by tsegismont** [comment]
Fixed by #678
